client/src/components/LocationSelect.tsx

```diff
@@ -1,25 +1,43 @@
-import * as React from 'react';
-import { Select } from 'antd';
-import { CITIES } from 'services/reference-data';
+import React from 'react';
+import AlgoliaPlaces from 'algolia-places-react';
+
+export type Location = {
+  cityName: string;
+  countryName: string;
+};
 
 type Props = {
-  [key: string]: any;
+  onChange: Function;
+  location: Location;
 };
 
-type LocationOption = { id?: string; name: string };
-const defaultOption: LocationOption = { id: undefined, name: 'Other' };
-const options: LocationOption[] = [defaultOption].concat(CITIES);
+const getDefaultValue = (location: Location): string =>
+  location && location.countryName !== null ? `${location.countryName}, ${location.cityName}` : '';
+
+export function LocationSelect(props: Props) {
+  const discardLocation = () => {
+    props.onChange(null);
+  };
 
-export class LocationSelect extends React.Component<Props> {
-  render() {
-    return (
-      <Select showSearch optionFilterProp="children" placeholder="Select..." {...this.props}>
-        {options.map((location, i) => (
-          <Select.Option key={location.id || i} value={location.id!}>
-            {location.name}
-          </Select.Option>
-        ))}
-      </Select>
-    );
-  }
+  return (
+    <AlgoliaPlaces
+      options={{
+        appId: 'plYRMXVHA4VI',
+        apiKey: 'c3457dc71fd196231949ed98db03f119',
+        language: 'en',
+        type: 'city',
+      }}
+      defaultValue={getDefaultValue(props.location)}
+      onSuggestions={discardLocation}
+      onClear={discardLocation}
+      onChange={({ suggestion }: any) => {
+        const location = {
+          countryName: suggestion.country,
+          cityName: suggestion.name,
+        } as Location;
+        props.onChange(location);
+      }}
+      placeholder="Write an address here"
```

Comment:
i think we don't need address, only city. I think it's better to say "Please enter your city" to be consistent with what we display in other controls.